### PR TITLE
Added localization support to name/description elements

### PR DESCRIPTION
**Pull request by jamesjhedley**
+Hijacked data2mxl node module in order to fix a bug in their code.

Thanks to @sleroux for a large portion of this work.

Fixes blackberry/BB10-Webworks-Packager#122

### CODE REVIEW

File: lib/config-parser.js [hunk 1]
```diff
@@ -418,9 +418,14 @@ function validateConfig(widgetConfig) {
     check(widgetConfig.version, localize.translate("EXCEPTION_INVALID_VERSION"))
         .notNull()
         .regex("^[0-9]{1,3}([.][0-9]{1,3}){2,3}$");
-    check(widgetConfig.name, localize.translate("EXCEPTION_INVALID_NAME")).notEmpty();
-    check(widgetConfig.author, localize.translate("EXCEPTION_INVALID_AUTHOR")).notNull();
 
+    for (var prop in widgetConfig.name) {
```
Comment (jeffheifetz) [inline lib/config-parser.js:422]: Shouldn't this be a forEach?

Comment (jamesjhedley) [inline lib/config-parser.js:422]: This is not actually an Array, so forEach is not available.

**Comment by nukulb** [comment]
@tracyli - can you test this soon as this needs to be in for tomorrows build?

**Comment by tracyli** [comment]
tested in mac with package build http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-twg_next-name-localization/4/: 
1.'name' can change when localization language changes; 
2.for 'description', I only checked the META-INF in the bar file: there're different bbr file generated for description with different language.

**Comment by jamesjhedley** [comment]
@tracyli Can you please test @EricLeiLi change as part of this pull request. Shouldn't need to retest my stuff.

**Comment by EricLeiLi** [comment]
@tracyli checkout blackberry#210 for details of packageLocale. You can contact me directly if you prefer.

**Comment by EricLeiLi** [comment]
@tracyli  I fixed the unit test. Let's wait for the next CI.

File: lib/config-parser.js [hunk 2]
```diff
@@ -492,6 +498,77 @@ function validateConfig(widgetConfig) {
...
+
+        if (attribs) {
+            language = attribs['xml:lang'] || 'default';
+            widgetConfig[tag][language] = tagElement['#'];
+        } else {
+            widgetConfig[tag]['default'] = tagElement;
+        }
+    }
+
+    if (tagData instanceof Array) {
```
Comment (jeffheifetz) [inline lib/config-parser.js]: Use Array.isArray

**Comment by jeffheifetz** [comment]
This requires a microsite issue

File: lib/config-parser.js [hunk 3]
```diff
@@ -492,6 +498,77 @@ function validateConfig(widgetConfig) {
...
+}
+
+function collectAndCombineLocales(locales, localeElements) {
+    var locale;
+
+    if (localeElements) {
+        for (locale in localeElements) {
+            if (localeElements.hasOwnProperty(locale)) {
+                //Don't add default locale and don't add locale if it already exists
+                if (locale !== 'default' && locales.indexOf(locale) === -1) {
```
Comment (jeffheifetz) [inline lib/config-parser.js]: Please make 'default' a constant

File: lib/config-parser.js (hunk 1 above)
Comment (jeffheifetz) [inline lib/config-parser.js:422]: OK then

**Comment by jamesjhedley** [comment]
Removed @EricLeiLi commit.
Implemented code review changes.
Rebased.

@jeffheifetz Can you r+ this before we get @tracyli to retest.

**Comment by jeffheifetz** [comment]
Jake lint fails for me

**Comment by tracyli** [comment]
tested in winxp and mac with http://ci0000003863287:8080/hudson/job/BB10-Webworks-Packager-twg_next-name-localization/12/: smoke test got passed;'name' can change when localization language changes;
for 'description', I only checked the META-INF in the bar file: there're different bbr file generated for description with different language.

**Comment by jeffheifetz** [comment]
r+

**Pull request closed without merging**